auto_tune_vllm/core/study_controller.py

```
@@ -387,14 +387,27 @@ def run_optimization(
+        # Require explicit, positive concurrency specification
+        if max_concurrent is None:
+            msg = (
+                "❌ --max-concurrent is required to prevent GPU memory conflicts!\n\n"
+                "Add to your YAML config:\n"
+                "  optimization:\n"
+                "    max_concurrent: 2  # Match your GPU count\n\n"
+                "Or use CLI: --max-concurrent 2"
+            )
+            raise ValueError(msg)
+        if max_concurrent < 1:
+            raise ValueError("--max-concurrent must be >= 1")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)